boards/toradex/colibri_imx7d/doc/index.rst

```diff
@@ -221,6 +221,193 @@ configured in the Zephyr compilation:
+         compatible = "shared-dma-pool";
+         reg = <0x90001000 0x1000>;
+         no-map;
+         phandle = <0xcd>; #--> biggest phandle +3
+      };
+
+      cm4tcmcode@7f8000 {
+         compatible = "shared-dma-pool";
+         reg = <0x7f8000 0x8000>;
+         no-map;
+         phandle = <0xce>; #--> biggest phandle +4
+      };
+
+      cm4sramcode@900000 {
+         compatible = "shared-dma-pool";
+         reg = <0x900000 0x40000>;
+         no-map;
+         phandle = <0xcf>; #--> biggest phandle +5
+      };
+
+      cm4reserved@8ff00000 {
+         compatible = "shared-dma-pool";
+         reg = <0x8ff00000 0x100000>;
+         no-map;
+         phandle = <0xd0>; #--> biggest phandle +6
+      };
+   };
+
+   imx7d-cm4 {
+      compatible = "fsl,imx7d-cm4";
+      mbox-names = "tx\0rx\0rxdb";
+      mboxes = <0xbd 0x00 0x00 0xbd 0x01 0x00 0xbd 0x03 0x00>; #--> MU_A phandle (0xbd)
+      memory-region = <0xcb 0xcc 0xcd 0xce 0xcf 0xd0>; #--> All the previously defined phandles
+      syscon = <0x32>; #--> phandle for the reset-controller
+      clocks = <0x01 0x42>;
+   };
+
+   #Recompile the dts into a dtb
+   dtc -I dts -O dtb -f imx7d-m4.dts -o imx7d-m4.dtb
+
+   #Copy the new dtb to /boot on the Colibri IMX7 board
+   #Start in U-boot and update the device-tree
+   setenv fdtfile imx7d-m4.dtb
+   saveenv
+   boot
+
+When the OS has finished booting with your new device tree you can enable
+the drivers and start the M4 core.
+
+.. code-block:: console
+
+   #Copy zephyr_openamp_rsc_table.elf to /lib/firmware on your board
+   modprobe imx_rproc
+   modprobe virtio_rpmsg_bus
+   modprobe rpmsg_tty
+
+   #Request RPROC to load the M4 image
+   echo stop > /sys/class/remoteproc/remoteproc0/state
+   echo zephyr_openamp_rsc_table.elf > /sys/class/remoteproc/remoteproc0/firmware
+   echo start > /sys/class/remoteproc/remoteproc0/state
+
+   #dmesg will detail the boot process:
+   dmesg
+   [  497.120499] remoteproc remoteproc0: stopped remote processor imx-rproc
+   [  497.138938] remoteproc remoteproc0: powering up imx-rproc
+   [  497.168735] remoteproc remoteproc0: Booting fw image zephyr_openamp_rsc_table.elf, size 1267076
+   [  497.184826] rproc-virtio rproc-virtio.1.auto: assigned reserved memory node vdev0buffer0@90002000
+   [  497.221395] virtio_rpmsg_bus virtio0: rpmsg host is online
+   [  497.233806] virtio_rpmsg_bus virtio0: creating channel rpmsg-tty addr 0x400
+   [  497.236666] rproc-virtio rproc-virtio.1.auto: registered virtio0 (type 7)
+   [  497.259822] remoteproc remoteproc0: remote processor imx-rproc is now up
+   [  497.293913] virtio_rpmsg_bus virtio0: creating channel rpmsg-client-sample addr 0x401
+   [  497.308388] rpmsg_client_sample virtio0.rpmsg-client-sample.-1.1025: new channel: 0x401 -> 0x401!
+   [  497.337969] virtio_rpmsg_bus virtio0: creating channel rpmsg-tty addr 0x402
+
+   ls /dev | grep ttyRPMSG
+   ttyRPMSG0 -> used for zephyr shell interface
+   ttyRPMSG1 -> used for sample interface
```

Comment:
The below should provide better syntax coloring

```suggestion
   #Copy zephyr_openamp_rsc_table.elf to /lib/firmware on your board
   $ modprobe imx_rproc
   $ modprobe virtio_rpmsg_bus
   $ modprobe rpmsg_tty

   #Request RPROC to load the M4 image
   $ echo stop > /sys/class/remoteproc/remoteproc0/state
   $ echo zephyr_openamp_rsc_table.elf > /sys/class/remoteproc/remoteproc0/firmware
   $ echo start > /sys/class/remoteproc/remoteproc0/state

   #dmesg will detail the boot process:
   $ dmesg
   [  497.120499] remoteproc remoteproc0: stopped remote processor imx-rproc
   [  497.138938] remoteproc remoteproc0: powering up imx-rproc
   [  497.168735] remoteproc remoteproc0: Booting fw image zephyr_openamp_rsc_table.elf, size 1267076
   [  497.184826] rproc-virtio rproc-virtio.1.auto: assigned reserved memory node vdev0buffer0@90002000
   [  497.221395] virtio_rpmsg_bus virtio0: rpmsg host is online
   [  497.233806] virtio_rpmsg_bus virtio0: creating channel rpmsg-tty addr 0x400
   [  497.236666] rproc-virtio rproc-virtio.1.auto: registered virtio0 (type 7)
   [  497.259822] remoteproc remoteproc0: remote processor imx-rproc is now up
   [  497.293913] virtio_rpmsg_bus virtio0: creating channel rpmsg-client-sample addr 0x401
   [  497.308388] rpmsg_client_sample virtio0.rpmsg-client-sample.-1.1025: new channel: 0x401 -> 0x401!
   [  497.337969] virtio_rpmsg_bus virtio0: creating channel rpmsg-tty addr 0x402

   $ ls /dev | grep ttyRPMSG
   ttyRPMSG0 -> used for zephyr shell interface
   ttyRPMSG1 -> used for sample interface
```

Reply:
I'm gonna try that and rebuild the docs to see how it looks :-)

Reply:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/67622881/0c5597f0-d200-43a1-8396-4171c17d6437)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/67622881/baa96a39-5a15-419e-9ed7-4b61c7acee40)

Does indeed provide syntax coloring ;)

Reply:
Solved